provider/dummy/environs.go

```diff
@@ -981,7 +981,7 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (*environs.St
 	series := args.Tools.OneSeries()
 
 	idString := fmt.Sprintf("%s-%d", e.name, estate.maxId)
-	addrs := network.NewAddresses(idString+".dns", "127.0.0.1")
+	addrs := network.NewAddresses(idString+".dns", "127.0.0.1", "::1")
```

Comment:
This file is already using 'localhost' as the mongo address. Here I'm adding the IPv6 address to the list of addresses, rather than using 'localhost' and having two hostnames, but no properly scoped, addresses.